class MyException extends Exception { }

// Half of the time this function fails and thorws an exception
void myFunction() {
    if( rand() < 0.5) throw new MyException()
}

try {
    // This code block might throw an exception
    println "Try block"
    myFunction()
} catch(MyException e) {
    // This block is executes only if the `try` block
    // throws an exception of type `MyException`
    println "MyException was thrown:\n$e"
} finally {
    // This code is ALWAYS executed
    println "Finally, we are finished"
} 